### PR TITLE
refactor: Upload full token

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |       Refactor Upload with full token.    |
| 🇨🇳 Chinese |      -     |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/upload/style/index.tsx [hunk 1]
```diff
@@ -38,18 +41,23 @@ const genBaseStyle: GenerateStyle<FullToken<'Upload'>> = token => {
...
-  }),
-);
+export default genComponentStyleHook('Upload', token => {
+  const { fontSizeHeading3, fontSizeBase, lineHeight, lineWidth, controlHeightLG } = token;
+  const listItemHeightSM = Math.round(fontSizeBase * lineHeight);
+
+  const uploadToken = mergeToken<UploadToken>(token, {
+    uploadThumbnailSize: fontSizeHeading3 * 2,
+    uploadProgressOffset: listItemHeightSM / 2 + lineWidth,
+    uploadPicCardSize: controlHeightLG * 2.55,
```
[the remark this answers is not in the document]
Comment (zombieJ) [reply components/upload/style/index.tsx:51]: 嗯，是的。但是先不管了。反正 `controlHeightLG` 是基本单元。

File: components/upload/demo/picture-style.md [hunk 2]
```diff
@@ -19,15 +19,21 @@ import { UploadOutlined } from '@ant-design/icons';
 
 const fileList = [
   {
-    uid: '-1',
+    uid: '0',
```
[the remark this answers is not in the document]
Comment (zombieJ) [reply components/upload/demo/picture-style.md:22]: 多一个 uploading 的样式校验，因为结构上已经和 v4 不一样了。所以胆子可以大点~

File: components/upload/style/picture.tsx [hunk 3]
```diff
@@ -25,19 +24,17 @@ const genPictureStyle: GenerateStyle<FullToken<'Upload'>> = token => {
...
-            // FIXME: upload token
-            opacity: 0.8,
             flex: 'none',
             overflow: 'hidden',
             textOverflow: 'ellipsis',
             whiteSpace: 'nowrap',
 
             [iconCls]: {
-              fontSize: uploadPictureThumbnailSize / 2 + 2,
+              fontSize: uploadThumbnailSize / 2 + 2,
```
[the remark this answers is not in the document]
Comment (zombieJ) [reply components/upload/style/picture.tsx:37]: 26 怎么都除不尽，我再想想

**Review by MadCcc — APPROVED** [review]

File: components/upload/style/index.tsx [hunk 4]
```diff
@@ -38,18 +41,23 @@ const genBaseStyle: GenerateStyle<FullToken<'Upload'>> = token => {
...
-    genDraggerStyle(token),
-    genPictureStyle(token),
-    genPictureCardStyle(token),
-    genListStyle(token),
-    genMotionStyle(token),
-    genRtlStyle(token),
-  ],
-  token => ({
-    uploadActionsColor: token.colorTextSecondary,
-  }),
```
[the remark this answers is not in the document]
Comment (zombieJ) [reply components/upload/style/index.tsx:54]: 嗯。下面的 hover 用的 `colorText`。这边就直接用 `colorTextSecondary` 了。

[the remark this answers is not in the document]
Comment (MadCcc) [reply components/upload/style/index.tsx:54]: 目前收敛的规则尽量严格，能从 token 派生的都没有放在 ComponentToken 里

**Pull request merged**